### PR TITLE
Add support for SICD 1.2

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

File: six/modules/c++/six.sicd/source/ComplexXMLControl.cpp [hunk 1]
```diff
@@ -117,6 +118,13 @@ ComplexXMLControl::getParser(const std::string& version) const
             // parser itself doesn't - so we can simply reuse the 1.0.1 parser.
             parser.reset(new ComplexXMLParser101(version, mLog));
         }
+        else if (minorVersion == "1" && patchVersion == "0")
+        {
+            // From a SIX standpoint, 1.2.0 is identical to 1.1.0. The changes
+            // get hanlded by XML_DATA_CONTENT.c. We just have a new version
```
Comment (asylvest) [inline six/modules/c++/six.sicd/source/ComplexXMLControl.cpp]: 'hanlded' typo.  I'd actually reword the second sentence to "The only difference between 1.2 and 1.1 is some rules with how FTITLE can be populated, so we can again reuse the 1.0.1 parser"

File: six/modules/c++/six.sicd/source/ComplexXMLControl.cpp [hunk 2]
```diff
@@ -117,6 +118,13 @@ ComplexXMLControl::getParser(const std::string& version) const
             // parser itself doesn't - so we can simply reuse the 1.0.1 parser.
             parser.reset(new ComplexXMLParser101(version, mLog));
         }
+        else if (minorVersion == "1" && patchVersion == "0")
```
Comment (asylvest) [inline six/modules/c++/six.sicd/source/ComplexXMLControl.cpp]: Need `minorVersion == "2"` right?

**Comment by asylvest** [comment]
Can you also add whatever's needed to the tests so that they're testing SICD 1.2?

**Comment by JonathanMeans** [comment]
The tests are updated, and the typos are fixed.

**Comment by asylvest** [comment]
Looks good - thanks

**Pull request merged**